### PR TITLE
Add operating system to user agent string

### LINKED ISSUE #183: Incorrect operating systems category from mac chrome
Issue tracker is **ONLY** used for reporting bugs. New features should be discussed on our slack channel.

## Expected Behavior

Activity sourced from chrome extension on Mac should be categorized into "Mac" on dashboard "Operating Systems".

## Current Behavior

Categorized into "Linux" which I didn't use recently.
It's confirmed by click the "Linux" part on dashboard, all is the website url shown up.

## Possible Solution

Check the api to read system info from chrome extension.

## Context (Environment)

macOS 13.2.1 (22D68)
Chrome Version 111.0.5563.110 (Official Build) (x86_64)

### PR DESCRIPTION
**Pull request by alanhamlett**
Fixes #183.

### CODE REVIEW

File: src/core/WakaTimeCore.ts [hunk 1]
```diff
@@ -348,6 +349,14 @@ class WakaTimeCore {
     return payload;
   }
 
+  getOperatingSystem(): Promise<string> {
+    return new Promise((resolve) => {
+      chrome.runtime.getPlatformInfo(function(info) {
```
Comment (jvelezpo) [inline src/core/WakaTimeCore.ts:354]: I am not sure this works for FF
We might want to use the browser pollyfill for this

File: src/core/WakaTimeCore.ts [hunk 2]
```diff
@@ -348,6 +349,14 @@ class WakaTimeCore {
     return payload;
   }
 
+  getOperatingSystem(): Promise<string> {
+    return new Promise((resolve) => {
```
Comment (jvelezpo) [inline src/core/WakaTimeCore.ts:353]: Please don't wrap code in Promise functions, Let's keep using async/await over it

Comment (alanhamlett) [reply src/core/WakaTimeCore.ts:353]: I did this so we could use `await` when calling `chrome.runtime.getPlatformInfo` to turn it into an `async` function.

File: src/core/WakaTimeCore.ts (hunk 1 above)
Comment (jvelezpo) [reply src/core/WakaTimeCore.ts:354]: Testing `chrome`  in FF, it is not defined 
image

**Review by jvelezpo — APPROVED** [review]

**Pull request merged**